packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/project/services/DiagramImporterUpdateService.java

```diff
@@ -0,0 +1,213 @@
+        Diagram oldRepresentation = (Diagram) representationImportData.representation();
+        var editingContext = this.editingContextSearchService.findById(editingContextId);
+        if (editingContext.isPresent()) {
+            var newRepresentation = this.representationSearchService.findById(editingContext.get(), newRepresentationId, Diagram.class);
+            var diagramDescription = this.representationDescriptionSearchService.findById(editingContext.get(), oldRepresentation.getDescriptionId())
+                    .filter(DiagramDescription.class::isInstance)
+                    .map(DiagramDescription.class::cast);
+
+            if (diagramDescription.isPresent() && newRepresentation.isPresent()) {
+                var oldLayoutData = oldRepresentation.getLayoutData();
+                var diagramContext = new DiagramContext(newRepresentation.get());
+                Map<String, String> nodeElementOldNewIds = new HashMap<>();
+                Map<String, String> edgeElementOldNewIds = new HashMap<>();
+                oldRepresentation.getNodes().forEach(oldNode -> {
+                    var nodeDescription = diagramDescription.get().getNodeDescriptions().stream().filter(nodeDesc -> nodeDesc.getId().equals(oldNode.getDescriptionId())).findFirst();
+                    if (nodeDescription.isPresent() && nodeDescription.get().getSynchronizationPolicy().equals(SynchronizationPolicy.UNSYNCHRONIZED)) {
+                        var viewCreationRequest = ViewCreationRequest.newViewCreationRequest()
+                                .parentElementId(newRepresentation.get().getId())
+                                .targetObjectId(semanticElementsIdMappings.get(oldNode.getTargetObjectId()))
+                                .descriptionId(oldNode.getDescriptionId())
+                                .containmentKind(NodeContainmentKind.CHILD_NODE)
+                                .build();
+                        diagramContext.getViewCreationRequests().add(viewCreationRequest);
+                    }
+
+                    var oldNodeId = oldNode.getId();
+                    var newNodeId = this.computeNodeId(newRepresentation.get().getId(), oldNode.getDescriptionId(), NodeContainmentKind.CHILD_NODE, semanticElementsIdMappings.get(oldNode.getTargetObjectId()));
+                    nodeElementOldNewIds.put(oldNodeId, newNodeId);
+
+                    oldNode.getChildNodes().forEach(childNode ->
+                        handleChildren(diagramDescription.get(), childNode, newNodeId, NodeContainmentKind.CHILD_NODE, semanticElementsIdMappings, nodeElementOldNewIds, diagramContext));
+                    oldNode.getBorderNodes().forEach(childNode ->
+                        handleChildren(diagramDescription.get(), childNode, newNodeId, NodeContainmentKind.BORDER_NODE, semanticElementsIdMappings, nodeElementOldNewIds, diagramContext));
+                });
+
+                AtomicInteger count = new AtomicInteger();
```

Comment:
I won't block this PR so close from the release for that but using some concurrent data structure for such a simple need seems odd.